File: 백준/Silver/9996. 한국이 그리울 땐 서버에 접속하지/한국이 그리울 땐 서버에 접속하지.py
N = int(input())
pattern = input()

arr = [input() for _ in range(N)]

star = pattern.find("*")
prefix = pattern[:star]
surfix = pattern[star+1:]

for i in arr:
  if len(pattern) - 1 <= len(i) and i.startswith(prefix) and i.endswith(surfix):
    print("DA")
  else:
    print("NE")

File: 백준/Silver/9996. 한국이 그리울 땐 서버에 접속하지/한국이 그리울 땐 서버에 접속하지.java
import java.util.*;

public class Main {

  static int N;
  static String pattern;
  static int starIdx;
  
  public static void main(String[] args) {

    Scanner in = new Scanner(System.in);

    N = Integer.parseInt(in.nextLine());
    
    pattern = in.nextLine();
    starIdx = pattern.indexOf("*");

    String start = pattern.substring(0, starIdx);
    String end = pattern.substring(starIdx + 1);

    for (int i = 0; i < N; i++) {
      String file = in.nextLine();

      if (file.length() >= pattern.length() - 1 &&
        file.startsWith(start) && file.endsWith(end)) {
        System.out.println("DA");
        }
      else {
        System.out.println("NE");
      }
    }
    in.close();
    
  }
}